In [36]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

#### Above, find required imports per formatting best practices; imports address the requirements of Q3 which is noted below.
##### Exercise questions 1 - 3 relate to ex-Jupyter functions required for setup and are noted below for reference:
Q1. Create a 'data' folder in your local project repository. </br> 
Q2. Download these two CSV files and place them in the data folder: </br>  
&ensp; http://data.un.org/Data.aspx?d=WDI&f=Indicator_Code%3aNY.GDP.PCAP.PP.KD </br>  
&emsp; - rename the file to `gdp_percapita.csv` </br>  
&ensp; Internet http://data.un.org/Data.aspx?d=ITU&f=ind1Code%3aI99H </br>  
&emsp; - rename the file to `internet_use.csv` </br>  
&ensp; Create a `notebooks` folder and launch a Jupyter Notebook in this folder.</br>  
Q3. Import the required packages with their customary aliases as follows: </br>  
    `import pandas as pd` </br>   
    `import numpy as np`  </br>  
    `import matplotlib.pyplot as plt`  </br>  
    `import seaborn as sns`

#### Q4. Using the pandas read_csv() method, read the GDP dataset into your notebook as a DataFrame called gdp_df. Take a look at the first few and last few rows to familiarize yourself with what is contained in this dataset.

In [37]:
#dataframe creation
gdp_df = pd.read_csv('../data/gdp_percapita.csv')

In [38]:
#dataframe preview
print(gdp_df.info)
print(gdp_df.dtypes)

<bound method DataFrame.info of      Country or Area  Year        Value  Value Footnotes
0        Afghanistan  2021  1517.016266              NaN
1        Afghanistan  2020  1968.341002              NaN
2        Afghanistan  2019  2079.921861              NaN
3        Afghanistan  2018  2060.698973              NaN
4        Afghanistan  2017  2096.093111              NaN
...              ...   ...          ...              ...
7657        Zimbabwe  1994  2670.106615              NaN
7658        Zimbabwe  1993  2458.783255              NaN
7659        Zimbabwe  1992  2468.278257              NaN
7660        Zimbabwe  1991  2781.787843              NaN
7661        Zimbabwe  1990  2704.757299              NaN

[7662 rows x 4 columns]>
Country or Area     object
Year                 int64
Value              float64
Value Footnotes    float64
dtype: object


#### Q5. How many rows and columns are in gdp_df? What are the data types of each column? 
#### A5.
  &ensp; SHAPE: </br>
    &emsp; 7662 rows; 4 columns
</br>
  &ensp; TYPE: </br>
    &emsp; Country or Area: object </br>
    &emsp; Year: int64 </br>
    &emsp; Value: float64 </br>
    &emsp; Value Footnotes: float64

#### Q6. Drop the `Value Footnotes` column and rename the remaining three to 'Country', 'Year', and 'GDP_Per_Capita'.

In [39]:
#Drop Value Footnotes
gdp_df = gdp_df.drop ('Value Footnotes', axis=1)
#rename remaining columns
gdp_df.columns = ['Country', 'Year', 'GDP_Per_Capita']
gdp_df

Country  Year  GDP_Per_Capita
0     Afghanistan  2021     1517.016266
1     Afghanistan  2020     1968.341002
2     Afghanistan  2019     2079.921861
3     Afghanistan  2018     2060.698973
4     Afghanistan  2017     2096.093111
...           ...   ...             ...
7657     Zimbabwe  1994     2670.106615
7658     Zimbabwe  1993     2458.783255
7659     Zimbabwe  1992     2468.278257
7660     Zimbabwe  1991     2781.787843
7661     Zimbabwe  1990     2704.757299

[7662 rows x 3 columns]

#### Q7. How many countries have data for all years? Which countries are missing many (at least one) years of data? Look at the number of observations per year. What do you notice?

In [51]:
#determine min year
print('Min Year: ' + str(gdp_df['Year'].min()))
#determine max year
print('Max Year: ' + str(gdp_df['Year'].max()))
print('Number of years in dataset: ' + str((gdp_df['Year'].max - str(gdp_df['Year'].min)))
#do a count - any countries that don't have 33 distinct years would have missing years

SyntaxError: incomplete input (1898912823.py, line 6)